src/pages/iou/steps/MoneyRequestAmountForm.js

```diff
@@ -123,9 +123,9 @@ function MoneyRequestAmountForm({amount, currency, isEditing, forwardedRef, onCu
             return;
         }
         initializeAmount(amount);
-        // we want to update the state only when the amount is changed
+        // we want to re-initialize the state only when the selected tab changes
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [amount]);
+    }, [selectedTab]);
```

Comment:
Was there any reason for removing `amount` from dependency?
It caused https://github.com/Expensify/App/issues/33125 and https://github.com/Expensify/App/issues/33123

Reply:
We discussed this internally and thought it made sense to only initialize the amount when the tab changes since that's the intended behavior. It seems like the code was coupled elsewhere though which caused the regressions above.

Reply:
ok, so it's fine to add this back?
```
    }, [amount, selectedTab]);
```

Reply:
Yea, I think it's ok as long as the amount still resets when the tab is switched